tests/Doctrine/Tests/DBAL/Psr11ConnectionRegistryTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ConnectionRegistry;
+use Doctrine\DBAL\Psr11ConnectionRegistry;
+use InvalidArgumentException;
+use PHPUnit\Framework\MockObject\Stub;
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerInterface;
+use function array_keys;
+
+class Psr11ConnectionRegistryTest extends TestCase
+{
+    /** @var ConnectionRegistry */
+    private $registry;
+
+    /** @var array<string, Connection&Stub> */
+    private $connections;
+
+    protected function setUp() : void
+    {
+        /** @var Connection&Stub $fooConnection */
+        $fooConnection = $this->createStub(Connection::class);
+        /** @var Connection&Stub $barConnection */
+        $barConnection = $this->createStub(Connection::class);
+
+        $this->connections = [
+            'foo' => $fooConnection,
```

Comment:
What would be a realistic name? `doctrine.dbal.connection.foo`? If yes maybe there should a decorating implementation that has a `$prefix` argument so that we can still use `getConnection('foo')?

Reply:
For now I don't think we need to worry about prefixes.

In the Bundle the name will be whatever you configure on the config. So `foo` would be a possible name.

Don't confuse it with the Symfony service name for the connections. Those will be like `doctrine.dbal.foo_connection` for example.

Also see

https://github.com/doctrine/dbal-bundle/pull/1/files#diff-6648b62ba0e1a03e4f23ed65e8efb9f5R70

Reply:
@greg0ire DoctrineDBALBundle won't wire this with the full Symfony container, but with a ServiceLocator, mapping connection names to services, not service ids to services. So this is indeed realistic (well, as far as `foo` is a realistic name for your connection)

Reply:
Oh ok I get it, thanks!